### PR TITLE
First round of updates:

### PR DESCRIPTION
**Pull request by sheyd**
1. Put in case logic to swap between environments. URL per prod or
thor gets set correctly. Now just have to make sure I set the DB
tables names correctly per env.
2. Cleaned up a bunch of the DB code to take strings as their proper
values and escaped securely. Also cleaned up query sanitizer to return
None types that DB cursor now properly sets to NULL value, not as
as string with value "NULL".
3. Add SSL connectivity for a bit more security over the wire.

Gem of a DB extraction class available here:
http://softwareengineering.stackexchange.com/questions/200522/how-to-deal-with-database-connections-in-a-python-library-module

#### What's this PR do?
1. Put in case logic to swap between environments. URL per prod or
thor gets set correctly. Now just have to make sure I set the DB
tables names correctly per env.
2. Cleaned up a bunch of the DB code to take strings as their proper
values and escaped securely. Also cleaned up query sanitizer to return
None types that DB cursor now properly sets to NULL value, not as
as string with value "NULL".
3. Add SSL connectivity for a bit more security over the wire.

Gem of a DB extraction class available here:
http://softwareengineering.stackexchange.com/questions/200522/how-to-deal-with-database-connections-in-a-python-library-module

#### Where should the reviewer start?
Any of the files attached.

#### How should this be manually tested?
Manually tested against Thor.

### CODE REVIEW

File: etl-scripts/northstar_to_user_table.py [hunk 1]
```diff
@@ -20,48 +20,68 @@
...
+        northstar_env_url = 'https://northstar.dosomething.org'
+    elif sys.argv[1] == 'thor':
+        northstar_env_url = 'https://northstar-thor.dosomething.org'
+    else:
+        print("Please provide a working Northstar environment.")
+        sys.exit(0)
+    i = int(sys.argv[2])
+else:
+    print("Sorry, please specify proper arguments.")
+"""Determine environment, and page to start from.
```
Comment (mshmsh5000) [inline etl-scripts/northstar_to_user_table.py]: Seems to me that a simpler and less error-prone approach would be always to require both arguments. The usage text will be easier to understand, and the invoker will always have to be explicit about page & environment.

File: etl-scripts/northstar_to_user_table.py [hunk 2]
```diff
@@ -20,48 +20,68 @@
...
+ns_fetcher = NorthstarScraper(northstar_env_url)
+
+
+def to_string(base_value):
+    """Converts to string and replaces None values with empty values."""
+    if base_value is None:
+        return None
+    else:
+        base_string = str(base_value)
+        strip_special_chars = re.sub(r'[()<>/"\'\\]', '', base_string)
```
Comment (mshmsh5000) [inline etl-scripts/northstar_to_user_table.py]: Is the exhaustive list of characters you want to strip out? Curious how you derived this set.

File: etl-scripts/northstar_to_user_table.py [hunk 3]
```diff
@@ -74,13 +94,12 @@ def to_string(base_value):
...
                                             moco_current_status,\
                                             moco_source_detail)\
-                                            VALUES(\"{0}\",\"{1}\",{2},\"{3}\",\
-                                            \"{4}\",\"{5}\",\"{6}\",\"{7}\",\
-                                            \"{8}\",\"{9}\",\"{10}\",\"{11}\",\
-                                            \"{12}\",\"{13}\",\"{14}\",\"{15}\"\
-                                            ,NULL,NULL,\"{16}\",\"{17}\",\
-                                            \"{18}\")".format(
-                                            to_string(user['id']),
+                                            VALUES(%s,%s,%s,%s,\
```
Comment (mshmsh5000) [inline etl-scripts/northstar_to_user_table.py]: This looks much cleaner.

File: etl-scripts/northstar_to_user_table.py [hunk 4]
```diff
@@ -20,48 +20,68 @@
...
 
-
+ca_settings = {'ca': '/home/quasar/rds-combined-ca-bundle.pem'}
 db = MySQLdb.connect(host=config.host,  # hostname
                      user=config.user,  # username
-                     passwd=config.pw)  # password
+                     passwd=config.pw,  # password
+                     use_unicode=True,  # Use unicode for queries.
+                     charset='utf8',    # Use UTF8 character set for queries.
+                     ssl=ca_settings)   # Connect using SSL
```
Comment (mshmsh5000) [inline etl-scripts/northstar_to_user_table.py]: Does this throw a useful error if, e.g., the file is missing/unreadable? Would want to be able to debug this quickly if connections were failing.

File: etl-scripts/thor_northstar_to_user_table.py [hunk 5]
```diff
@@ -20,47 +20,68 @@
...
+    else:
+        base_string = str(base_value)
+        strip_special_chars = re.sub(r'[()<>/"\'\\]', '', base_string)
+        return str(strip_special_chars)
 
 while nextPage is True:
     current_page = ns_fetcher.getUsers(100, i)
     for user in current_page:
-        query = "REPLACE INTO quasar.thor_users (northstar_id,\
+        cur.execute("REPLACE INTO quasar.thor_users (northstar_id,\
```
Comment (mshmsh5000) [inline etl-scripts/thor_northstar_to_user_table.py]: So table names like this still need to get altered per environment, right?

Comment (sheyd) [inline etl-scripts/thor_northstar_to_user_table.py]: @mshmsh5000 Yeah. I think I can do a simple `%s` substitution per environment, but haven't tested it yet. Probably will this morning and add to this PR.

File: etl-scripts/northstar_to_user_table.py (hunk 2 above)
Comment (sheyd) [inline etl-scripts/northstar_to_user_table.py]: Basically most use cases we've errored out on. I think we need a better general purpose parser, perhaps building on something like this: http://pathvalidate.readthedocs.io/en/latest/pages/examples/sanitize.html#replace-symbols. But I also want to do some more research in the data/security community to see if there's a good all-around parser, especially when dealing with UTF8 and international symbols.

File: etl-scripts/northstar_to_user_table.py (hunk 1 above)
Comment (sheyd) [inline etl-scripts/northstar_to_user_table.py]: Will modify and update this.

File: etl-scripts/northstar_to_user_table.py (hunk 4 above)
Comment (sheyd) [inline etl-scripts/northstar_to_user_table.py]: Yes, it will give SSL invalid file warning.

File: etl-scripts/northstar_to_user_table.py (hunk 2 above)
Comment (mshmsh5000) [inline etl-scripts/northstar_to_user_table.py]: Seems safe to start out with.

File: etl-scripts/thor_northstar_to_user_table.py (hunk 5 above)
Comment (sheyd) [inline etl-scripts/thor_northstar_to_user_table.py]: I've run into string substitution errors on string DB env, it's inserting an extra single-quote and breaking the string query. I have the code stashed and will push up another PR and create a backlog ticket to clean this up later. To get member count out sooner, going to pivot and focus on MailChimp scraping code. Merging in.

**Pull request merged**